shared/src/main/kotlin/org/javacs/kt/classpath/CachedClassPathResolver.kt

```diff
@@ -12,7 +12,7 @@ import org.jetbrains.exposed.sql.transactions.transaction
 import java.nio.file.Path
 import java.nio.file.Paths
 
-private const val MAX_PATH_LENGTH = 255
+private const val MAX_PATH_LENGTH = 65535
```

Comment:
I agree that the path limit should be longer than 255. Is the memory impact negligible here, given that this column length is used for every class path entry? I think we would do fine with a lower limit, most file systems probably don't even allow paths this long.

Perhaps

```suggestion
private const val MAX_PATH_LENGTH = 2047
```

I struggle to imagine a scenario where even that limit would be reached for a single class path entry.

Reply:
> Is the memory impact negligible here

I agree with your concern.

> I think we would do fine with a lower limit, most file systems probably don't even allow paths this long.

I agree. I've heard that the limit on path length in ext4 is 4096 (although it is configurable).
I feel 2047 is enough for most cases.